Formula/c/cargo-c.rb

```diff
@@ -24,6 +24,10 @@ class CargoC < Formula
   end
 
   depends_on "rust" => :build
+  # The `cargo` crate requires http2, which `curl-config` from macOS reports to
+  # be missing despite its presence.
+  # Try switching to `uses_from_macos` when that's resolved.
+  depends_on "curl"
```

Comment:
If this is only needed on Sonoma: can we do it only on Sonoma instead?

Also: this seems like a good candidate for a `uses_from_macos before:` (like `since:`) if we ever need more of these.

Reply:
These are actually two separate problems.

The `CoreServices` issue in the PR description, which caused our bottling attempts fail, only happened on Sonoma.

`curl-config` reporting missing `http2`, however, happens across macOS versions. We can tell that because the `curl-sys` crate would automatically link against system libcurl if http2 is found to be supported [^1], while the linkage summaries in the previous version bump show that none of the binaries from the current bottles, from Big Sur to Ventura, have system libcurl linkage [^2].

When `curl-config` does not http2 support, `curl-sys` would build a bundled libcurl instead (and also libnghttp2, as a dependency). We don't like vendored dependencies, so I think we may want to use the brewed `curl` on all macOS versions. In that way the `curl-config` from our own `curl` is used, which does report http2 support and make the `curl-sys` crate happy.

[^1]: https://github.com/alexcrichton/curl-rust/blob/4bf8eb2a87c8f9e4a6552a1fcf4f9695d7a600f3/curl-sys/build.rs#L27
[^2]: https://github.com/Homebrew/homebrew-core/actions/runs/5991437609

Reply:
> Also: this seems like a good candidate for a `uses_from_macos before:` (like `since:`) if we ever need more of these.

Yea, I agree 👍

Reply:
> When `curl-config` does not http2 support, `curl-sys` would build a bundled libcurl instead (and also libnghttp2, as a dependency). We don't like vendored dependencies, so I think we may want to use the brewed `curl` on all macOS versions. In that way the `curl-config` from our own `curl` is used, which does report http2 support and make the `curl-sys` crate happy.

Gotcha. Agreed 👍🏻